src/App.js

```diff
@@ -10,18 +13,35 @@ class App extends Component {
     super();
 
     this.state = {
-      currentUser: 'Nathaniel'
+      currentUser: '',
+      id: ''
+      //maybe consider Nan lets do research
     }
   }
 
+  logInUser = (user) => {
+    console.log(user)
+    this.setState({currentUser: user.user.name, id: user.user.id})
+    // this.setState(user.user) potential refactor later
+  }
+
+  logOutUser = () => {
+    this.setState({currentUser: '', id: ''})
+  }
+
   render() {
     return (
       <main>
-        <Navbar currentUser={this.state.currentUser} />
-        <section class="movie-directory">
-          <h2>Top Rated Movies</h2>
-          <MovieMain />
-        </section>
+        <Navbar currentUser={this.state.currentUser} signOut = {this.logOutUser} />
+        <Switch>
+          <Route exact path="/" component={MovieMain}/>
+          <Route exact path="/sign-in" render={
+            () => { if (this.state.currentUser === 'Lucy'){
+              return <Redirect to="/"/>
+            } else {
+              return <SignIn logIn={this.logInUser}/>
+            }} }/>
+        </Switch>
```

Comment:
This switch is pretty awesome. I wonder if there is a way we can make it easier to read because it's a little hard to tell that the render is calling an anonymous function inside of it. Perhaps make a helper function instead? Our first time using it so not sure of the best path for it.